golem-worker-service-base/src/service/gateway/api_deployment.rs

```diff
@@ -624,13 +658,19 @@ where
 
             Err(ApiDeploymentError::ApiDeploymentConflict(site.clone()))
         } else {
+            // API definitions corresponding to the deployment
+            let existing_api_definitions = self.get_definitions_by_site(namespace, site).await?;
+
             self.deployment_repo
                 .delete(existing_deployment_records.clone())
                 .await?;
 
             self.set_undeployed_as_draft(existing_deployment_records)
                 .await?;
 
+            self.remove_component_constraints(existing_api_definitions, auth_ctx)
+                .await?;
+
```

Comment:
There are many instances like this - what happens if x runs and y fails. It's hard to squash everything into repo level transactions.

But that's a different topic. Not a concern of this PR.

Reply:
Let's leave a comment in the code for this so it's easier to find all the places with broken semantics later when we want to fix

Reply:
Sorry I missed this comment. 

These race conditions and transaction guarantee problems cannot be done just with Repository and any service layers. It's an architectural revisit. 

Not saying the current state is too bad, and I am pretty sure all of us the whole team knows it and aware of it - and there fore these localised comments will only confuse us - ending up thinking - only these commented places have issues and our solution becomes too specific and localised (wasting time solving it in a suboptimal way)